### PR TITLE
[bugfix] ensure cloning codes tensor before replacing bos and eos with the padding audio token.

### PR DESCRIPTION
**Pull request by XuesongYang**
1. call `.clone()` to avoid in-place modification since other places may reuse the input `codes` tensor.
2. disable `autocast` to `bf16` during codec to audio reconstruction. Verified that the reconstructed audio sounds much better when keeping fp32. For example, with the wrapper, `pred_audio.dtype` keeps 

*original info before applying this change*
```python
    510         self,
    511         logits,
    512         target_audio_codes,
    513         audio_codes_lens_target,
    514         context_audio_codes=None,
    515         context_audio_codes_lens=None,
    516     ):
    517         pred_audio_codes = self.logits_to_audio_codes(logits, audio_codes_lens_target)
    518         pred_audio, pred_audio_lens = self.codes_to_audio(pred_audio_codes, audio_codes_lens_target)
    519         target_audio, target_audio_lens = self.codes_to_audio(target_audio_codes, audio_codes_lens_target)
--> 520         import ipdb; ipdb.set_trace()

ipdb> x = self.logits_to_audio_codes(logits, audio_codes_lens_target)
ipdb> torch.equal(x[0], pred_audio_codes[0])
False
ipdb> audio_codes_lens_target[0]
tensor(82, device='cuda:0', dtype=torch.int32)
ipdb> x[0][:,81]
tensor([2047, 2047, 2047, 2047, 2047, 2047, 2047, 2047], device='cuda:0')
ipdb> pred_audio_codes[0][:,81]
tensor([0, 0, 0, 0, 0, 0, 0, 0], device='cuda:0')
ipdb> pred_audio_codes.dtype
torch.int64
ipdb> x.dtype
torch.int64
ipdb> pred_audio.dtype
torch.bfloat16
```
*info after applying this change*
```python
    520         context_audio_codes_lens=None,
    521     ):
    522         pred_audio_codes = self.logits_to_audio_codes(logits, audio_codes_lens_target)
    523         pred_audio, pred_audio_lens = self.codes_to_audio(pred_audio_codes, audio_codes_lens_target)
    524         target_audio, target_audio_lens = self.codes_to_audio(target_audio_codes, audio_codes_lens_target)
    525         import ipdb; ipdb.set_trace()
--> 526         context_audio, context_audio_lens = None, None
    527         if context_audio_codes is not None and context_audio_codes.shape[2] > 3:
    528             # > 3 ensures, it is a valid context audio tensor (and not dummy tensor used in text context)
    529             context_audio, context_audio_lens = self.codes_to_audio(context_audio_codes, context_audio_codes_lens)
    530

ipdb> pred_audio.dtype
torch.float32
ipdb> pred_audio_codes[0][:81]
tensor([2047, 2047, 2047, 2047, 2047, 2047, 2047, 2047], device='cuda:0')
```

### CODE REVIEW

**Comment by blisc** [comment]
Added this change in #57
